collectors/ebpf.plugin/README.md

```diff
@@ -130,87 +204,112 @@ By default, MD flush is disabled. To enable it, configure your
-software doesn't commonly call these functions directly, they are behind the system calls `open(2)`, `openat(2)`,
-and `close(2)`.
+To give metrics related to `open` and `close` events, instead of attaching kprobes for each syscall used to do these 
+events, the collector attaches `kprobes` for the common function used for syscalls:
+
+- [do_sys_open](https://0xax.gitbooks.io/linux-insides/content/SysCall/linux-syscall-5.html ): Internal function used to 
+   open files.
+- [do_sys_openat2](https://elixir.bootlin.com/linux/v5.6/source/fs/open.c#L1162): Function called from do_sys_open since 
+   version `5.6.0`.
+- [close_fd](https://www.mail-archive.com/linux-kernel@vger.kernel.org/msg2271761.html): Function used to close file 
+  descriptor since kernel `5.11.0`.
+- `__close_fd`: Function used to close files before version `5.11.0`.
 
 #### File error
 
 This chart shows the number of times some software tried and failed to open or close a file descriptor.
 
-### VFS
-
-A [virtual file system](https://en.wikipedia.org/wiki/Virtual_file_system) (VFS) is a layer on top of regular
-filesystems. The functions presented inside this API are not used for filesystems, so it's possible that the charts in
-this section won't show _all_ the actions that occurred on your system.
-
-#### Deleted objects
+#### VFS Deleted objects 
 
 This chart monitors calls to `vfs_unlink`. This function is responsible for removing objects from the file system.
 
-#### IO
+#### VFS IO
 
 This chart shows the number of calls to the functions `vfs_read` and `vfs_write`.
 
-#### IO bytes
+#### VFS IO bytes
 
 This chart also monitors `vfs_read` and `vfs_write` but, instead of the number of calls, it shows the total amount of
 bytes read and written with these functions.
 
 The Agent displays the number of bytes written as negative because they are moving down to disk.
 
-#### IO errors
+#### VFS IO errors
 
 The Agent counts and shows the number of instances where a running program experiences a read or write error.
 
-#### Create
+#### VFS Create
 
 This chart shows the number of calls to `vfs_create`. This function is responsible for creating files.
 
-#### Synchronization
+#### VFS Synchronization
 
 This chart shows the number of calls to `vfs_fsync`. This function is responsible for calling `fsync(2)` or
 `fdatasync(2)` on a file. You can see more details in the Synchronization section.
 
-#### Open
+#### VFS Open
 
 This chart shows the number of calls to `vfs_open`. This function is responsible for opening files.
 
-### eBPF
+#### Directory Cache
 
-#### Process thread
+Metrics for directory cache are collected using kprobe for `lookup_fast`, because we are interested in the number of 
+times this function is accessed. On the other hand, for `d_lookup` we are not only interested in the number of times it 
+is accessed, but also in possible errors, so we need to attach a `kretprobe`. So, the following is used:
```

Comment:
Some rewording is needed here to avoid repetition: " So, the following is used:" e.g "For that reason, the following is used: " @DShreve2 any thoughts?

Reply:
Agree with this. Need to be aware of using "so... so... "repetitively. "For that reason" is better in the last instance.